### ISSUE TITLE
New rules for "blood moon" cards

### ISSUE THREAD

**Issue by JayDi85** [magefree/mage#4202]
[Issue from reddit](https://www.reddit.com/r/XMage/comments/7fiwec/blood_moon_rules/).

Is xMage work correct with new rule changes for "blood moon" cards? You can read about changes [on that page](https://magic.wizards.com/en/articles/archive/news/ixalan-comprehensive-rules-changes-2017-09-28) (search by "614.12, 614.16d").

> So this rule's changing. Now, to determine what replacement effects will apply to a permanent entering the battlefield, you look at continuous effects that modify that permanent's characteristics coming from the permanent itself along with those that already exist and will affect that permanent. Dimir Aqueduct enters the battlefield untapped under Blood Moon. Humility causes creatures with modular to enter without their +1/+1 counters. Metallic Mimic can distribute +1/+1 counters happily to your army of Ferrets under Arcane Adaptation.

**Comment by theelk801** [comment]
Yeah, we know about these but they're hard to implement given that they need to look forward in time in order to apply. I'll try to figure something out.

**Comment by JayDi85** [comment]
Card review with many possible rules and situations from mtg judges:
http://cantrip.ru/en/mtg-cards/Blood-Moon.shtml

**Comment by LevelX2** [comment]
from jeffwadsworth:
https://magic.wizards.com/en/articles/archive/news/ixalan-comprehensive-rules-changes-2017-09-28

Essentially, any effect that would affect the card coming into the battlefield gets applied before it actually enters the battlefield. So a 1/1 white creature would be a 2/2 when it enters the battlefield with a Crusade in play, not a 1/1 that gets boosted to 2/2. It is a very fundamental change.

**Comment by LevelX2** [comment]
## The core of the new behaviour 614.12
Now, to determine what **replacement effects** will apply to a permanent entering the battlefield, you look at **continuous effects** that modify **that permanent's characteristics** coming from the **permanent itself** along with **those that already exist** and will affect that permanent. 

## Characteristic-defining ability
* 604.3. Some static abilities are characteristic-defining abilities. A characteristic-defining ability conveys information about an object’s characteristics that would normally be found elsewhere on that object (such as in its mana cost, type line, or power/toughness box) or overrides information found elsewhere on that object. Characteristic-defining abilities function in all zones. They also function outside the game.
* 604.3a A static ability is a characteristic-defining ability if it meets the following criteria: (1) It defines an object’s colors, subtypes, power, or toughness; (2) it is printed on the card it affects, it was granted to the token it affects by the effect that created the token, or it was acquired by the object it affects as the result of a copy effect or text-changing effect; (3) it does not directly affect the characteristics of any other objects; (4) it is not an ability that an object grants to itself; and (5) it does not set the values of such characteristics only if certain conditions are met.

## What has to happen if a permanent enters the battlefield

1.  Determine which continuous effects from the permanent itself modify that permanent's characteristics.
2. Determine which already existing continuous effects modify that permanent's characteristics.
3. Apply that continuous effects to the permanent before entering.
4. Let the permanent enter the battlefield.

Beside the yet not clear way how to do that, is that list and the order of the list correct?
Does this list miss something that has to be taken into account also?

Any suggestions/additions welcome!
